core/spec.md

```diff
@@ -934,6 +939,75 @@ of the existing entity. Then the existing entity would be deleted.
 - Examples:
   - `/endpoints/ep1`
 
+##### `xbase` Attribute
+
+- Type: XID
+- Description: This attribute is used to have the current entity "inherit" or
+  "extend" another entity. This is most useful when a shared common definition
+  is needed across multiple instances, and only a small set of custom changes
+  are needed.
+
+  The `xbase` attribute MAY only be used on Group, Resource or Version
+  instances and MUST reference an entity of the same model type as the
+  current entity.
+
+  The act of inheriting another entity entails having the referenced entity's
+  attributes visible as part of the current entity. This only applies to
+  attributes at the top-level of the entity, and each of those attributes
+  MUST be visible in its entirety.
+
+  Any attributes in the target entity that are also present in the current
```

Comment:
do bottoms up instead of top down.
Follow all xbases, and then work backwards overlaying each up you go up.